tests/test_create_points.py

```diff
@@ -14,3 +29,64 @@ def test_help():
         "Create a dataset of interpolated points along OpenStreetMap roads."
         in result.output
     )
+
+
+def test_filter_by_highway_type_error():
+    bad_df = gpd.GeoDataFrame(columns=["not_highway"])
+    with pytest.raises(Exception):
+        filter_by_highway_type(bad_df, [])
+
+
+def test_filter_by_highway_type():
+    highway_vals = np.random.choice(DEFAULT_HIGHWAY_VALUES_TO_KEEP, size=10).tolist()
+    drop_highway_vals = ["drop_value_1", "drop_value_2", "drop_value_3"]
+    df = gpd.GeoDataFrame({"highway": highway_vals + drop_highway_vals})
+    filtered_df = filter_by_highway_type(df)
+    assert all(val in DEFAULT_HIGHWAY_VALUES_TO_KEEP for val in filtered_df["highway"])
+
+
+def test_interpolate_along_line():
+    test_line = LineString([[-9532658.679, 5154379.822], [-9534693.722, 5154037.319]])
+    test_dist = 20
+    new_coords = interpolate_along_line(test_line, test_dist)
+    assert isinstance(new_coords, MultiPoint)
+    assert len(new_coords.geoms) == int(test_line.length / test_dist)
+
+
+@pytest.mark.parametrize("invalid_value", [None, geometry.Point((1, 1))])
+def test_create_points_error(invalid_value):
+    test_df = gpd.read_file("tests/assets/test_gdf.shp").head()
+    test_df.loc[0, "geometry"] = invalid_value
+    with pytest.raises(Exception):
```

Comment:
If you know what exception will be raised by an invalid value (ValueError?), put that here instead of the generic `Exception` so that we still error on unexpected errors